packages/integration-tests/src/test-e2e/deployment.test.ts

```diff
@@ -124,6 +126,37 @@ void describe(
 
             await testProject.assertPostDeployment();
           });
+
+          void it(`[${testProjectCreator.name}] generates client config with provided directory and format`, async () => {
+            const configOutDir = 'temp';
+
+            // create directory for client config
+            const tempPath = path.join(
+              testProject.projectDirPath,
+              configOutDir
+            );
+            await fs.mkdir(tempPath);
+
+            // test generating all formats in temp directory
+            for (const format of Object.values(ClientConfigFormat)) {
+              await amplifyCli(
+                [
+                  'sandbox',
+                  '--config-out-dir',
+                  configOutDir,
+                  '--config-format',
+                  format,
+                ],
+                testProject.projectDirPath
+              )
+                .do(waitForSandboxDeploymentToPrintTotalTime())
+                .do(interruptSandbox())
+                .do(rejectCleanupSandbox())
+                .run();
```

Comment:
the ask in the issue was to add tests fro `amplify generate config` not sandbox options.
this should likely become part of `pipeline-deploy` test somewhere in this file since it needs some deployment first.

(which would be faster at runtime btw).

Reply:
Would it be useful to keep one test case to make sure `--config-out-dir` and `--config-format` are working as expected?

Reply:
I'd say no, we've got pretty solid coverage in unit tests for that and likely tests for `generate config` are going to catch integration problems. We should hold until we get signal that we need this e2e test.

As for `generate config`, it's presumably one of our core commands that has zero e2e test coverage. Hence the need to run it at least in one of tests.

Reply:
Done in [019bc64](https://github.com/aws-amplify/amplify-backend/pull/682/commits/019bc647c3d54dcf6cc472b2bef215de2661c4fc)